readthedocs/builds/reporting.py

```diff
@@ -0,0 +1,31 @@
+from django.template.loader import render_to_string
+
+from readthedocs.builds.models import Build
+from readthedocs.filetreediff import get_diff
+
+
+def get_build_report(build: Build):
+    project = build.project
+    base_version = project.addons.options_base_version or project.get_latest_version()
+    if not base_version:
+        return None
+
+    diff = get_diff(
+        current_version=build.version,
+        base_version=base_version,
+    )
+
+    if not diff:
+        return None
```

Comment:
We should post the comment even if the diff is not available. It's a setting and it can be disabled by the user.

Reply:
The diff is none if there isn't a manifest to get the diff from. If the diff doesn't have changes, we still return a diff object.